Analysis/Producer.py

```diff
@@ -30,27 +30,35 @@ def defineAllP4(self):
             self.defineP4(f"b{idx+1}")
             self.defineP4(f"tau{idx+1}_seedingJet")
 
+    def defineRegions(self):
+        tau2_iso_var = f"tau2_idDeepTau{self.deepTauYear()}{self.deepTauVersion}VSjet"
+        self.df = self.df.Define("region_A", f"tau1_charge*tau2_charge < 0 && {tau2_iso_var} >= {Utilities.WorkingPointsTauVSjet.Medium.value}")
+        self.df = self.df.Define("region_B", f"tau1_charge*tau2_charge > 0 && {tau2_iso_var} >= {Utilities.WorkingPointsTauVSjet.Medium.value}")
+        self.df = self.df.Define("region_C", f"tau1_charge*tau2_charge < 0 && {tau2_iso_var} < {Utilities.WorkingPointsTauVSjet.Medium.value} && {tau2_iso_var} >= {Utilities.WorkingPointsTauVSjet.VVVLoose.value}")
+        self.df = self.df.Define("region_D", f"tau1_charge*tau2_charge > 0 && {tau2_iso_var} < {Utilities.WorkingPointsTauVSjet.Medium.value} && {tau2_iso_var} >= {Utilities.WorkingPointsTauVSjet.VVVLoose.value}")
+
+
     def createInvMass(self):
         self.df = self.df.Define("tautau_m_vis", "(tau1_p4+tau2_p4).M()")
         self.df = self.df.Define("bb_m_vis", "(b1_p4+b2_p4).M()")
         self.df = self.df.Define("bbtautau_mass", "(b1_p4+b2_p4+tau1_p4+tau2_p4).M()")
         self.df = self.df.Define("dR_tautau", 'ROOT::Math::VectorUtil::DeltaR(tau1_p4, tau2_p4)')
 
+    def deepTauYear(self):
+        return deepTauYears[self.deepTauVersion]
+
     def __init__(self, df):
         self.df = df
         self.deepTauVersion = 'v2p1'
-        self.deepTauYear = deepTauYears[self.deepTauVersion]
         self.defineAllP4()
         self.createInvMass()
+        self.defineRegions()
 
 
     def skimAnatuple(self):
-        for tau_idx in [1,2]:
-            self.df = self.df.Filter(f'tau{tau_idx}_idDeepTau{self.deepTauYear}{self.deepTauVersion}VSjet >= {Utilities.WorkingPointsTauVSjet.Medium.value}')
-        self.df = self.df.Filter('tau1_charge * tau2_charge <0')
+        self.df = self.df.Filter(f'tau1_idDeepTau{self.deepTauYear()}{self.deepTauVersion}VSjet >= {Utilities.WorkingPointsTauVSjet.Medium.value}')
```

Comment:
```suggestion
```
this is should be applied at the baseline level. isn't it?

Reply:
I am not sure. I put it to be safe